src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs

```diff
@@ -182,17 +182,30 @@ protected PatchResult ExecuteOnDocument(DocumentsOperationContext context, strin
 
                             result.Status = PatchStatus.Created;
                         }
-                        else if (DocumentCompare.IsEqualTo(originalDoc, modifiedDocument, 
-                                    DocumentCompare.DocumentCompareOptions.MergeMetadataAndThrowOnAttachmentModification) != DocumentCompareResult.Equal)
+                        else 
                         {
-                            Debug.Assert(originalDocument != null);
-                            if (_isTest == false || run.PutOrDeleteCalled)
+                            DocumentCompareResult compareResult;
+                            try
+                            {
+                                compareResult = DocumentCompare.IsEqualTo(originalDoc, modifiedDocument,
+                                    DocumentCompare.DocumentCompareOptions.MergeMetadataAndThrowOnAttachmentModification);
+                            }
+                            catch (InvalidOperationException ioe)
                             {
-                                putResult = _database.DocumentsStorage.Put(context, originalDocument.Id,
-                                    originalDocument.ChangeVector, modifiedDocument, null, null, originalDocument.Flags);
+                                throw new InvalidOperationException("Illegal modifications of '@attachments' detected for document: " + id);
```

Comment:
Better to just state that there was an error processing this and use the `ioe` as inner exception

Reply:
Done